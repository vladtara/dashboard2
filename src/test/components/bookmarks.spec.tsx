import { render } from "@testing-library/react";
import BookmarkList, {
  Bookmark,
  BookmarkGroup,
  IBookmarkGroupProps,
  IBookmarkListProps,
} from "../../components/bookmarks";

const bookmarkGroupProps: IBookmarkGroupProps = {
  name: "Test Group",
  items: [
    {
      name: "Bookmark Test",
      url: "#",
    },
  ],
};

const bookmarkListProps: IBookmarkListProps = {
  groups: [bookmarkGroupProps, bookmarkGroupProps],
};

it("tests rendering of Bookmark with newTab=true", () => {
  let props = bookmarkGroupProps.items[0];
  const { baseElement } = render(
    <Bookmark name={props.name} url={props.url} newTab={true} />,
  );

  expect(baseElement).toMatchSnapshot();
});

it("tests rendering of Bookmark with newTab=false", () => {
  let props = bookmarkGroupProps.items[0];
  const { baseElement } = render(
    <Bookmark name={props.name} url={props.url} newTab={false} />,
  );

  expect(baseElement).toMatchSnapshot();
});

it("tests rendering of Bookmark without newTab", () => {
  let props = bookmarkGroupProps.items[0];
  const { baseElement } = render(
    <Bookmark name={props.name} url={props.url} />,
  );

  expect(baseElement).toMatchSnapshot();
});

it("tests rendering of BookmarkGroup", () => {
  const { baseElement } = render(
    <BookmarkGroup
      name={bookmarkGroupProps.name}
      items={bookmarkGroupProps.items}
    />,
  );

  expect(baseElement).toMatchSnapshot();
});

it("tests rendering of BookmarkList", () => {
  const { baseElement } = render(
    <BookmarkList groups={bookmarkListProps.groups} />,
  );

  expect(baseElement).toMatchSnapshot();
});
