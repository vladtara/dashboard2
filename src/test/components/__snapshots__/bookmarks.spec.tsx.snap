// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests rendering of Bookmark with newTab=false 1`] = `
<body>
  <div>
    <a
      class="sc-iBkjds ANOjH"
      href="#"
    >
      Bookmark Test
    </a>
  </div>
</body>
`;

exports[`tests rendering of Bookmark with newTab=true 1`] = `
<body>
  <div>
    <a
      class="sc-iBkjds ANOjH"
      href="#"
      rel="noopener noreferrer"
      target="_blank"
    >
      Bookmark Test
    </a>
  </div>
</body>
`;

exports[`tests rendering of Bookmark without newTab 1`] = `
<body>
  <div>
    <a
      class="sc-iBkjds ANOjH"
      href="#"
    >
      Bookmark Test
    </a>
  </div>
</body>
`;

exports[`tests rendering of BookmarkGroup 1`] = `
<body>
  <div>
    <li
      class="sc-eCYdqJ jcnfPp"
    >
      <div
        class="sc-gKXOVf kDgtUh"
      >
        <h3
          class="sc-dkzDqf dzkwjK"
        >
          Test Group
        </h3>
        <a
          class="sc-iBkjds ANOjH"
          href="#"
        >
          Bookmark Test
        </a>
      </div>
    </li>
  </div>
</body>
`;

exports[`tests rendering of BookmarkList 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ kimCm"
    >
      <h2
        class="sc-gsnTZi jiVNCT"
      >
        Bookmarks
      </h2>
      <ul
        class="sc-hKMtZM jEBHIA"
      >
        <li
          class="sc-eCYdqJ jcnfPp"
        >
          <div
            class="sc-gKXOVf kDgtUh"
          >
            <h3
              class="sc-dkzDqf dzkwjK"
            >
              Test Group
            </h3>
            <a
              class="sc-iBkjds ANOjH"
              href="#"
            >
              Bookmark Test
            </a>
          </div>
        </li>
        <li
          class="sc-eCYdqJ jcnfPp"
        >
          <div
            class="sc-gKXOVf kDgtUh"
          >
            <h3
              class="sc-dkzDqf dzkwjK"
            >
              Test Group
            </h3>
            <a
              class="sc-iBkjds ANOjH"
              href="#"
            >
              Bookmark Test
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
